pkg/controller/controllercmd/builder.go

```diff
@@ -328,8 +349,19 @@ func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstru
 	// ensure blocking TCP connections don't block the leader election
 	leaderConfig := rest.CopyConfig(protoConfig)
 	leaderConfig.Timeout = b.leaderElection.RenewDeadline.Duration
+	var leaderElection leaderelection.LeaderElectionConfig
+
+	// default to configmapsleases for leader election
+	if b.leaderElectionResourceLock == "" || b.leaderElectionResourceLock == ConfigMapsLeasesResourceLock {
+		leaderElection, err = leaderelectionconverter.ToLeaderElectionWithConfigmapLease(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity)
+	} else if b.leaderElectionResourceLock == LeasesResourceLock {
+		leaderElection, err = leaderelectionconverter.ToLeaderElectionWithLease(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity)
+	} else if b.leaderElectionResourceLock == EndpointsLeasesResourceLock {
+		leaderElection, err = leaderelectionconverter.ToLeaderElectionWithEndpointLease(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity)
+	} else {
+		return fmt.Errorf("%w: %s", ErrInvalidResourceLock, b.leaderElectionResourceLock)
+	}
```

Comment:
it would be cleaner to have a switch here

Reply:
Good suggestion. Thanks!